### ISSUE TITLE
tensorflow-gpu 2.0.1 hangs mid-epoch

### ISSUE THREAD

**Issue by ben-arnao** [tensorflow/tensorflow#37450]
<em>Please make sure that this is a bug. As per our
[GitHub Policy](https://github.com/tensorflow/tensorflow/blob/master/ISSUES.md),
we only address code/doc bugs, performance issues, feature requests and
build/installation issues on GitHub. tag:bug_template</em>

**System information** 
- Have I written custom code (as opposed to using a stock
example script provided in TensorFlow): The only modification i have made is that i've forked ReduceLRonPlateau and use this modified version. The only change is that i reduce the weight decay along with the learning rate. Seems highly unlikely this is relevant.

ReduceLRonPlateau
```
 old_lr = float(K.get_value(self.model.optimizer.lr))
                    old_wd = float(K.get_value(self.model.optimizer.weight_decay))
                    if old_lr > self.min_lr:
                        new_lr = old_lr * self.factor
                        new_wd = old_wd * self.factor
                        new_lr = max(new_lr, self.min_lr)
                        K.set_value(self.model.optimizer.weight_decay, new_wd)
                        K.set_value(self.model.optimizer.lr, new_lr)
```

- OS Platform and Distribution (e.g.,
Linux Ubuntu 16.04):  Windows 10
- TensorFlow installed from (source or
binary): - TensorFlow version (use command below):  pip
- Python version: - Bazel
version (if compiling from source):  3.6
- GCC/Compiler version (if compiling from
source): 
- CUDA/cuDNN version: - GPU model and memory: 1060 6GB

You can collect some of this information using our environment capture
[script](https://github.com/tensorflow/tensorflow/tree/master/tools/tf_env_collect.sh)
You can also obtain the TensorFlow version with: 1. TF 1.0: `python -c "import
tensorflow as tf; print(tf.GIT_VERSION, tf.VERSION)"` 2. TF 2.0: `python -c
"import tensorflow as tf; print(tf.version.GIT_VERSION, tf.version.VERSION)"`

**Describe the current behavior**

Upon calling model.fit() TF will hang at seemingly random points. (5 epochs in, mid epoch for example after we've trained on 20% of our batches). I am doing hyper parameter optimization, so sometimes a trial will complete fully, other times it will hang indefinitely. I was having no issues originally when doing a regression problem. This issue only started happenign when i switched to categorical classification.

My output node is 3 units because i have 3 class. Using a softmax function and built in categorical cross entropy. Those are the only changes from when it was working to now.

Tried changing batch size, does not seem to have affect. Not sure if i can see a pattern if certain parameter configurations cause the fit to fail.

I notice the CPU spikes during a stall.

**Describe the expected behavior**

TF does not stall or at least provided some information as to why it is stalled.

**Standalone code to reproduce the issue** 
Provide a reproducible test case that is the bare minimum necessary to generate
the problem. If possible, please share a link to Colab/Jupyter/any notebook.

**Other info / logs** Include any logs or source code that would be helpful to
diagnose the problem. If including tracebacks, please include the full
traceback. Large logs and files should be attached.

**Comment by ben-arnao** [comment]
Upgrading to 2.1 seemed to fix whatever bug this is.

**Comment by ben-arnao** [comment]
Issue still happens with tensorflow 2.1

**Comment by gadagashwini** [comment]
@ben-arnao, Please provide the complete standalone code to replicate the reported issue. Thanks!

**Comment by liuxingbaoyu** [comment]
Do you have a situation that occupies two cores?

**Comment by gadagashwini** [comment]
@ben-arnao, Please post the code snippet to analyze the issue. Thanks!

**Comment by gadagashwini** [comment]
Automatically closing due to lack of recent activity. Please update the issue when new information becomes available, and we will reopen the issue. Thanks!
